src/librustc_trans/mir/block.rs

```diff
@@ -95,17 +96,31 @@ impl<'bcx, 'tcx> MirContext<'bcx, 'tcx> {
                     adt::trans_get_discr(bcx, &repr, discr_lvalue.llval, None, true)
                 );
 
-                // The else branch of the Switch can't be hit, so branch to an unreachable
-                // instruction so LLVM knows that
-                let unreachable_blk = self.unreachable_block();
-                let switch = bcx.switch(discr, unreachable_blk.llbb, targets.len());
+                let mut bb_hist = FnvHashMap();
```

Comment:
You should add a FIXME here to check for the validity of such transformation. While currently the otherwise branch is always unreachable (and contains a panic), this might not be the case after a bunch of various optimisation passes are added to the MIR thus making this transformation invalid.